lib/src/common/async_extensions.dart

```diff
@@ -0,0 +1,16 @@
+import 'dart:async';
+
+/// Extensions for [Completer]
+extension CompleterExtensions<T> on Completer<T> {
+  void completeIfNotCompleted([FutureOr<T> value]) {
+    if (!isCompleted) {
+      complete(value);
+    }
+  }
+
+  void completeErrorIfNotCompleted(Object error, [StackTrace stackTrace]) {
+    if (!isCompleted) {
```

Comment:
I wonder if there's a possibility these could mask errors. Like, if I'm calling something with an error, I'm not sure that silently discarding it part of the time is always desirable.

Reply:
If it's already completed, it'll just result in a different, uncaught error, this time at the caller instead of consumer. So yes, it could very well mask additional errors, but at least the first one was sent to consumers. I think the trade-off is worth it. Consumers who disagree can always choose the default API.